### PR TITLE
Feat: Added single product detail page

### PR DESCRIPTION
**Pull request by nishasen**
It contains the detail of single product.
image

### CODE REVIEW

File: diagon-alley/src/Components/DetailedProduct/DetailedProduct.module.css [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    height: 20rem;
+}
+
+.product_desc {
+    background-color: hsla(0, 0%, 98%, 0.7);
+    padding: 1rem;
+    width: 18rem;
+    display: flex;
+    flex-direction: column;
+    gap: 0.4rem;
```
Comment (rdshinde) [inline diagon-alley/src/Components/DetailedProduct/DetailedProduct.module.css:22]: if it is less than 10px try to use pixel itself instead of rem.

**Review by rdshinde — APPROVED** [review]
Nicely Done. 
Just a small suggestion if spacing is less than 10px and it is not relative try to use pixel itself as a unit, I think you are using less than 1rem many time.

File: diagon-alley/src/Components/DetailedProduct/DetailedProduct.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    buttonBorder={true} 
+                    size={"large"} 
+                    className={style.changed_button_primary}
+                    onClick={()=>navigate('../cart', { replace: true })}/>
+            :
+            <Button buttonText={"🛒 Add to cart"} 
+                    buttonBorder={true} 
+                    size={"large"} 
+                    onClick={()=>addItemToCartHandler()}/>}
+
```
Comment (sadathshariff) [inline diagon-alley/src/Components/DetailedProduct/DetailedProduct.jsx:75]: Nice bit of work here

**Review by sadathshariff — APPROVED** [review]
The code looks clean and like how you are passing the ProductName in the Toast

File: diagon-alley/src/Components/DetailedProduct/DetailedProduct.module.css (hunk 1 above)
Comment (nishasen) [reply diagon-alley/src/Components/DetailedProduct/DetailedProduct.module.css:22]: I have taken fontsize as 16px, that is 1rem = 16px, so took it this way

**Pull request merged**